app/src/main/java/swati4star/createpdf/fragment/ZipToPdfFragment.java

```diff
@@ -0,0 +1,158 @@
+                Constants.pdfDirectory + Constants.tempDirectory;
+
+        try {
+            fileInputStream = new FileInputStream(mPath);
+            ZipInputStream zipInputStream = new ZipInputStream(new BufferedInputStream(fileInputStream));
+            ZipEntry zipEntry;
+            int folderPrefix = 0;
+
+            /* In case there are folders in the zip file and in those folders we have images
+             *  then we have possibility that file names in different folders are same.
+             *  In this case we will use folderPrefix so that all images are copied to temp
+             *  for every folder encountered we will increment folderPrefix by one and append
+             *  it to the the image name*/
+            while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+                String zipEntryName = zipEntry.getName().toLowerCase();
+
+                if (zipEntry.isDirectory()) {
+                    folderPrefix++;
+                } else if (zipEntryName.endsWith(".jpg") || zipEntryName.endsWith(".png")) {
+                    String newFileName = "/" + zipEntryName;
+                    int index = zipEntryName.lastIndexOf("/");
+                    /*index will be -1 when image is in just inside the zip
+                     * and not inside some folder*/
+                    if (index != -1)
+                        newFileName = zipEntryName.substring(index);
+                    if (folderPrefix != 0)
+                        newFileName = newFileName.replace("/", "/" + folderPrefix + "- ");
+                    File newFile = new File(dest + newFileName);
+                    imageUris.add(Uri.fromFile(newFile));
+
+                    byte[] buffer = new byte[BUFFER_SIZE];
+                    FileOutputStream fileOutputStream = new FileOutputStream(newFile);
+                    bufferedOutputStream = new BufferedOutputStream(fileOutputStream, BUFFER_SIZE);
+                    int count;
+
+                    while ((count = zipInputStream.read(buffer, 0, BUFFER_SIZE)) != -1) {
+                        bufferedOutputStream.write(buffer, 0, count);
+                    }
+
+                    bufferedOutputStream.flush();
+                    bufferedOutputStream.close();
+                }
+            }
+            zipInputStream.close();
+            extractingSnackbar.dismiss();
+
+            /*once we have extracted images out of zip, now we will pass
+             * image uri's and to main activity which will then be used to
+             * to start images to pdf fragment*/
+            ((MainActivity) getActivity()).convertZipToPdf(imageUris);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            extractingSnackbar.dismiss();
+            StringUtils.showSnackbar(mActivity, R.string.error_occurred);
+        } catch (IOException e) {
+            e.printStackTrace();
+            extractingSnackbar.dismiss();
```

Comment:
Add a `finally` block and move the `extractingSnackbar.dismiss()` to that section. It is present in all the three blocks.

Reply:
nice..will do it